src/main/java/teammates/ui/webapi/CreateAccountRequestAction.java

```diff
@@ -1,56 +1,209 @@
         }
+    }
 
-        String joinLink = accountRequestAttributes.getRegistrationUrl();
+    private String generateExistingAccountRequestErrorMessage(AccountRequestCreateIntent intent,
+                                                              String instructorEmail, String instructorInstitute) {
+        switch (intent) {
+        case ADMIN_CREATE:
+            AccountRequestAttributes accountRequest = logic.getAccountRequest(instructorEmail, instructorInstitute);
+            return "An account request already exists with status " + accountRequest.getStatus() + ".";
+        case PUBLIC_CREATE:
+            return "Oops, your submission is unsuccessful because an account request already exists."
+                    + " Please check if you have entered your personal information correctly."
+                    + " If you think this shouldn't happen, contact us at the email address given above.";
+        default:
+            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        }
+    }
 
-        EmailWrapper email = emailGenerator.generateNewInstructorAccountJoinEmail(
-                instructorEmail, instructorName, joinLink);
-        emailSender.sendEmail(email);
+    private boolean validateAccountRequestAndPopulateErrorResults(
+            AccountRequestCreateIntent intent, AccountRequestAttributes accountRequest,
+            AccountRequestCreateErrorResults errorResults) {
+        List<String> invalidityInfo = accountRequest.getInvalidityInfo();
+        if (invalidityInfo.isEmpty()) {
+            return true;
+        }
 
-        JoinLinkData output = new JoinLinkData(joinLink);
-        return new JsonResult(output);
+        for (String i : invalidityInfo) {
+            String ii;
+            if (i.startsWith(
+                    AccountRequestAttributes.generatePrefix(FieldValidator.PERSON_NAME_FIELD_NAME))) {
+                ii = StringHelper.removeFirstOccurrenceOfSubstring(i,
+                        AccountRequestAttributes.generatePrefix(FieldValidator.PERSON_NAME_FIELD_NAME));
+                errorResults.setInvalidNameMessage(ii);
+            } else if (i.startsWith(
+                    AccountRequestAttributes.generatePrefix(FieldValidator.ACCOUNT_REQUEST_INSTITUTE_NAME_FIELD_NAME))) {
+                ii = StringHelper.removeFirstOccurrenceOfSubstring(i,
+                        AccountRequestAttributes.generatePrefix(FieldValidator.ACCOUNT_REQUEST_INSTITUTE_NAME_FIELD_NAME));
+                errorResults.setInvalidInstituteMessage(ii);
+            } else if (i.startsWith(
+                    AccountRequestAttributes.generatePrefix(FieldValidator.ACCOUNT_REQUEST_COUNTRY_NAME_FIELD_NAME))) {
+                ii = StringHelper.removeFirstOccurrenceOfSubstring(i,
+                        AccountRequestAttributes.generatePrefix(FieldValidator.ACCOUNT_REQUEST_COUNTRY_NAME_FIELD_NAME));
+                errorResults.setInvalidCountryMessage(ii);
```

Comment:
None of these are needed. Errors are an exceptional case, because all of these should be validated before the form is sent on the frontend. Simply returning an error message if there are any errors is enough.

Reply:
No they are definitely needed. You can trust me on this. FE is also not doing much checking because it's hard to implement.

Reply:
For create account request, whatever returned from backend will be shown to users.